commercetools-sdk-base/src/main/java/io/sphere/sdk/utils/CompletableFutureUtils.java

```diff
@@ -64,49 +124,178 @@ private CompletableFutureUtils() {
         });
     }
 
+    /**
+     * Creates a {@link CompletionStage} which may recovered from an occurred error. Alias for {@link CompletionStage#exceptionally(Function)}.
+     * If the recovery function also requires to create a {@link CompletionStage} then use {@link #recoverWith(CompletionStage, Function)}.
+     *
+     * {@include.example io.sphere.sdk.utils.CompletableFutureUtilsTest#recoverFailure()}
+     *
+     * @param future future which may completes exceptionally
+     * @param f function how the exception should be handled to create a successfully completed future, but it also can throw exceptions
+     * @param <T> type of the value of the future
+     * @return a future which may recovered from errors
+     */
     public static <T> CompletionStage<T> recover(final CompletionStage<T> future, final Function<Throwable, ? extends T> f) {
         return future.exceptionally(f);
     }
 
-    public static <T> CompletionStage<T> recoverWith(final CompletionStage<T> future, final Function<? super Throwable, CompletionStage<T>> f) {
+    /**
+     * Creates a {@link CompletionStage} which may recovered from an occurred error.
```

Comment:
Actually, not. You can let an error happen anyway. So "which can be recovered if an error occurs". Because the developer has the power to recover it via this function. Boy, that's difficult phrasing :D